src/components/Post/Post.jsx

```diff
@@ -1,46 +1,215 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import "./Post.css";
-import { PostDropdown } from "../index";
+import { PostDropdown, Comment } from "../index";
+import { likePost, dislikePost } from "../../features/posts/postsSlice";
+import {
+  bookmarkPost,
+  removePostFromBookmark,
+} from "../../features/users/usersSlice";
+import {
+  getCommentsForPost,
+  createNewCommentToPost,
+  editCommentOfPost,
+} from "../../features/comments/commentsSlice";
+import { useDispatch, useSelector } from "react-redux";
+import TextareaAutosize from "react-textarea-autosize";
 
 function Post({ post, setIsPostEdit, setEditPostId }) {
   const { _id, content, username, name } = post;
+
+  const [isPostLikedBy, setIsPostLikedBy] = useState(false);
+  const [isPostBookmark, setIsPostBookmark] = useState(false);
   const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [showCommentBox, setShowCommentBox] = useState(false);
+  const [commentData, setCommentData] = useState({ text: "" });
+  const [editCommentData, setEditCommentData] = useState({
+    postId: "",
+    commentId: "",
+    isEditComment: false,
+  });
+
+  const token = localStorage.getItem("token");
+  const activeUsername = localStorage.getItem("username");
+
+  const dispatch = useDispatch();
+  const postObj = useSelector((state) => state.posts);
+  const bookmark = useSelector((state) => state.users.bookmarks);
+  const postsCommentsArray = useSelector((state) => state.comments.comments);
+  const comments = useSelector((state) =>
+    state.comments.comments.filter((post) => post._id == _id)
+  );
+
+  useEffect(() => {
+    const thisPost = postObj.posts.filter((post) => post._id == _id);
+    const isLikedPost = thisPost[0].likes.likedBy.some(
+      (user) => user.username == activeUsername
+    );
+
+    setIsPostLikedBy(isLikedPost);
+  }, [postObj]);
+
+  useEffect(() => {
+    const isBookmark = bookmark.some((user) => user._id == _id);
+    setIsPostBookmark(isBookmark);
+  }, [bookmark]);
+
```

Comment:
suggestion - This approach is also fine, 
isLikedby and isBookMarkedBy can be just functions which return boolean by processing values from slice 
ex - 
const isLikedPost = () => {
    const thisPost = postObj.posts.filter((post) => post._id == _id);
    return thisPost[0].likes.likedBy.some(
        (user) => user.username == activeUsername
    );
};

you can use isLikedPost() wherever required.
it automatically reruns when component re-render when slice updates

Reply:
Okay thanks , I will look into it